modules/uhttp/client/client.go

```diff
@@ -21,115 +21,53 @@
 package client
 
 import (
-	"io"
 	"net/http"
-	"net/url"
-	"strings"
 
 	"go.uber.org/fx"
 	"go.uber.org/fx/auth"
-	"go.uber.org/fx/config"
-
-	"golang.org/x/net/context/ctxhttp"
-)
-
-var (
-	_serviceName string
+	"go.uber.org/fx/internal/fxcontext"
 )
 
-// Client wraps around a http client
-type Client struct {
-	*http.Client
-	info                auth.CreateAuthInfo
-	filters             []Filter
-	defaultFiltersAdded bool
-}
-
-// New creates a new instance of uhttp Client
-func New(info auth.CreateAuthInfo, client *http.Client, filters ...Filter) *Client {
-	_serviceName = info.Config().Get(config.ApplicationIDKey).AsString()
-	filters = append(filters, tracingFilter(), authenticationFilter(info))
-	return &Client{
-		Client:              client,
-		info:                info,
-		filters:             filters,
-		defaultFiltersAdded: true,
+// New creates an http.Client that includes 2 extra filters: tracing and auth
+// they are going to be applied in following order: tracing, auth, remaining filters
+// and only if all of them passed the request is going to be send.
+func New(info auth.CreateAuthInfo, filters ...Filter) *http.Client {
+	extra := make([]Filter, len(filters)+2)
```

Comment:
Filter order matters, we want to have tracing first, then auth, and user filters next.